Sources/App/Core/Github.swift

```diff
@@ -67,94 +36,224 @@ enum Github {
+    }
+
+}
+
+
+// MARK: - GraphQL
+
+extension Github {
+
+    static let graphQLApiUri = URI(string: "https://api.github.com/graphql")
+
+    struct GraphQLQuery: Content {
+        var query: String
+    }
+
+    static func fetchResource<T: Decodable>(_ type: T.Type, client: Client, query: GraphQLQuery) -> EventLoopFuture<T> {
+        guard let token = Current.githubToken() else {
+            return client.eventLoop.future(error: Error.missingToken)
+        }
+        let headers = HTTPHeaders([("User-Agent", "SPI-Server"),
+                                   ("Authorization", "Bearer \(token)")])
+        return client.post(Self.graphQLApiUri, headers: headers) { req in
+            try req.content.encode(query)
+        }
+        .flatMap { response -> EventLoopFuture<ClientResponse> in
+            guard !isRateLimited(response) else {
+                return Current
+                    .reportError(client,
+                                 .critical,
+                                 AppError.metadataRequestFailed(nil,
+                                                                .tooManyRequests,
+                                                                Self.graphQLApiUri))
+                    .flatMap {
+                        client.eventLoop.future(error: Error.requestFailed(.tooManyRequests))
+                    }
+            }
+            return client.eventLoop.future(response)
+        }
+        .flatMapThrowing { response in
+            guard response.status == .ok else {
+                throw Error.requestFailed(response.status)
+            }
+            return try response.content.decode(T.self, using: decoder)
+        }
+    }
+
+    static func fetchMetadata(client: Client, owner: String, repository: String) -> EventLoopFuture<Metadata> {
+        struct Response: Decodable, Equatable {
+            var data: Metadata
+        }
+        return fetchResource(Response.self,
+                             client: client,
+                             query: Metadata.query(owner: owner, repository: repository))
+            .map(\.data)
+    }
+
+    static func fetchMetadata(client: Client, package: Package) -> EventLoopFuture<Metadata> {
+        do {
+            let (owner, name) = try parseOwnerName(url: package.url)
+            return fetchMetadata(client: client, owner: owner, repository: name)
+        } catch {
+            return client.eventLoop.future(error: error)
+        }
+    }
+
+}
+
+
+extension Github {
+
+    struct Metadata: Decodable, Equatable {
+        static func query(owner: String, repository: String) -> GraphQLQuery {
+            // Go to https://developer.github.com/v4/explorer/ to run query manually
+            GraphQLQuery(query: """
+                {
+                  repository(name: "\(repository)", owner: "\(owner)") {
+                    closedIssues: issues(states: CLOSED, first: 100, orderBy: {field: UPDATED_AT, direction: DESC}) {
```

Comment:
NB: because we can't sort by `CLOSED_AT` (doesn't exist as a sort field), we have to pull a batch and look for the latest client side. This is no worse than before: we also pulled in 100 issues via the REST call. The big advantage now is that we _only_ pull in the `closedAt` field and not 100 times the full issue payload, just to get the close date.